Cargo.toml

```
@@ -39,65 +39,19 @@ features = [
     "errhandlingapi"
 ]
 
-[target.i686-unknown-linux-gnu.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.x86_64-unknown-linux-gnu.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.arm-unknown-linux-gnueabihf.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.armv7-unknown-linux-gnueabihf.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.aarch64-unknown-linux-gnu.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.x86_64-unknown-dragonfly.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.armv6-unknown-freebsd.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.armv7-unknown-freebsd.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.aarch64-unknown-freebsd.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
-
-[target.x86_64-unknown-freebsd.dependencies]
-x11-dl = "2.18.3"
-wayland-client = "0.25"
-wayland-protocols = { version = "0.25", features = ["client"] }
-tempfile = "3.1.0"
+[features]
+default = ["wayland", "x11"]
+x11 = ["x11-dl"]
+wayland = ["wayland-client", "wayland-protocols", "wayland-cursor", "tempfile", "xkb", "xkbcommon-sys"]
+
+[target.'cfg(unix)'.dependencies]
```

```suggestion
[target.'cfg(not(target_os = "macos"))'.dependencies]
```

Pretty sure `unix` also counts macOS.

---

This will include redox, so we need to change it

---

Updated.